src/simulator/layer.h

```diff
@@ -46,10 +72,15 @@ internal bool layer_link(State* state,
                          Layer* layer, Layer* input_layer,
                          SynapseCls* cls, f32 weight); 
 
-// NOTE: to set inputs
-internal void layer_step(Layer* layer, u32 time);
-internal void layer_step_inject_current(Layer* layer, u32 time, f32* currents, u32 n_currents);
-internal void layer_step_force_spike(Layer* layer, u32 time, bool* spikes, u32 n_spikes);
+internal void layer_process_neurons(void* task); 
+
+internal void layer_step(Layer* layer, u32 time,
```

Comment:
We could combine the 3 functions into one but not sure, the code may get smaller but more complicated

Reply:
No, too ugly